src/handlers/get-thumbnail.js

```diff
@@ -30,22 +30,30 @@ function validateRequest(event) {
   return { id, aspect, size };
 }
 
-const getWorkThumbnail = async (id, aspect, size, event) => {
+const getThumbnail = async (id, aspect, size, event) => {
   const allowUnpublished = event.userToken.hasEntitlement(id);
   const allowPrivate = allowUnpublished || event.userToken.isReadingRoom();
 
-  const esResponse = await getWork(id, {
-    allowPrivate,
-    allowUnpublished,
-  });
-
-  if (esResponse.statusCode != 200) {
-    return opensearchResponse.transform(esResponse);
+  let esResponse;
+  let body;
+  let iiif_base;
+  if (event.rawPath.match(/\/collections\//)) {
```

Comment:
Do we not have to worry about unpublished and/or private collections?

Reply:
We do have to worry about private (reading room). Technically now a collection won't have an entitlement, but I suppose we could allow it some day. I don't think there is harm in leaving the code in there?

Reply:
But I have same comment as MBK about the code below - line 41 - it needs to pass the opts (sorry the way this was formatted I thought you were talking about lines 33-35

```
await getCollection(id, {
      allowPrivate,
      allowUnpublished,
    });
```

Reply:
Ok cool, I actually had that in but I took it out for some reason (maybe just because it was previously `getCollection(id)` though, can't remember the thought process). I'll update!